feat: astro:env stable (#9160)
withastro/docs: src/content/docs/en/reference/adapter-reference.mdx

    // emit a new physical file using your bundler
}
```

### envGetSecret

> To support [`astro:env` secrets](/en/guides/environment-variables/#getsecret), you need to update the `envGetSecret` adapter feature. Additionally, the `astro/env/setup` module allows you to provide an implementation for `getSecret()`. In your server entrypoint, call `setGetEnv()` as soon as possible:

Reviewer: Can we have a line explaining what this is before jumping into using it? Maybe matching the other entries (functionPerRoute was there before it was deleted in this branch. Maybe something like:

(I don't know if this should be `true`, or `"stable"` or what.)

``````suggestion
This is a feature to allow your adapter to retrieve secrets configured by users in `env.schema`.

Enable the feature by passing `true` to the adapter:

```js title="my-adapter.mjs" ins={9-11}
export default function createIntegration() {
  return {
    name: '@matthewp/my-adapter',
    hooks: {
      'astro:config:done': ({ setAdapter }) => {
        setAdapter({
          name: '@matthewp/my-adapter',
          serverEntrypoint: '@matthewp/my-adapter/server.js',
          adapterFeatures: {
              envGetSectret: true
          } 
        });
      },
    },
  };
}
```

The `astro/env/setup` module allows you to provide an implementation for `getSecret()`. In your server entrypoint, call `setGetEnv()` as soon as possible:
``````

```js ins={2,4}
import { App } from 'astro/app';
import { setGetEnv } from "astro/env/setup"

setGetEnv((key) => process.env[key])

export function createExports(manifest) {
  const app = new App(manifest);

  const handler = (event, context) => {
    // ...
  };

  return { handler };
}
```

> If you support secrets, calling `getSecret()` until `setGetEnv()` is called will throw a runtime error. This is important to keep in mind if your environment variables are tied to the request:

Reviewer: ```suggestion
If you support secrets, calling `getSecret()` before `setGetEnv()` is called will throw a runtime error. This is important to keep in mind if your environment variables are tied to the request:
```
should this be before?

    // emit a new physical file using your bundler
}
```

### envGetSecret

This is a feature to allow your adapter to retrieve secrets configured by users in `env.schema`.

> Enable the feature by passing any valid `SupportsKind` value to the adapter:

Reviewer: ```suggestion
Enable the feature by passing `envGetSectret: 'stable'` to the `adapterFeatures`:
```
Does this make more sense now?

Author: Well technically people could say it's in experimental, that's why i didn't want to say stable here

Reviewer: This is for the 5.0 beta docs though right? It's not experimental in these docs?

Author: I mean that's a different thing. Yes `astro:env` is stable but this option is for the adapter to tell its users if it supports `astro:env` well

Reviewer: AH, I was missing that part. OK, makes sense then!

```js title="my-adapter.mjs" ins={9-11}
export default function createIntegration() {
  return {
    name: '@matthewp/my-adapter',
    hooks: {
      'astro:config:done': ({ setAdapter }) => {
        setAdapter({
          name: '@matthewp/my-adapter',
          serverEntrypoint: '@matthewp/my-adapter/server.js',
          adapterFeatures: {
              envGetSectret: 'stable'
          } 
        });
      },
    },
  };
}
```

The `astro/env/setup` module allows you to provide an implementation for `getSecret()`. In your server entrypoint, call `setGetEnv()` as soon as possible:

```js ins={2,4}
import { App } from 'astro/app';
import { setGetEnv } from "astro/env/setup"

setGetEnv((key) => process.env[key])

export function createExports(manifest) {
  const app = new App(manifest);

  const handler = (event, context) => {
    // ...
  };

  return { handler };
}
```

> If you support secrets, calling `getSecret()` before `setGetEnv()` is called will throw a runtime error. This is important to keep in mind if your environment variables are tied to the request:

Reviewer: ```suggestion
If you support secrets, be sure to call `setGetEnv()` before `getSecret()` when your environment variables are tied to the request:
```
Is this correct to provide good guidance, in a postitive (ie. Do this!) way?
